File: backend-dotnet/Application/DTOs/UpdatePostDto.cs
using Application.Common;
using Domain.Enums;

namespace Application.DTOs
{
    public class UpdatePostDto
    {
        public string? Caption { get; set; }
        public UploadFile? Image { get; set; }
        public PostVisibility? Visibility { get; set; }
    }
}

File: backend-java/src/main/java/com/jerrygram/application/dtos/UpdatePostDto.java
package com.jerrygram.application.dtos;

import com.jerrygram.domain.enums.PostVisibility;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.web.multipart.MultipartFile;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class UpdatePostDto {
    private String caption;
    private PostVisibility visibility;
    private MultipartFile image;
}